src/WP_CLI/ParseGithubUrlInput.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace WP_CLI;
+
+final class ParseGithubUrlInput {
```

Comment:
Can we move the methods on this class to the existing `CommandWithUpgrade` class? I don't think we want to introduce it as a class others can use, etc.